qt/widgets/instrumentview/src/InstrumentActor.cpp

```diff
@@ -165,45 +205,47 @@ void InstrumentActor::setUpWorkspace(
   // set the ragged flag using a workspace validator
   auto wsValidator = Mantid::API::CommonBinsValidator();
   m_ragged = !wsValidator.isValid(sharedWorkspace).empty();
-
-  /// Keep the pointer to the detid2index map
-  m_detid2index_map = sharedWorkspace->getDetectorIDToWorkspaceIndexMap();
 }
 
-/** Used to set visibility of an actor corresponding to a particular component
-* When selecting a component in the InstrumentTreeWidget
-*
-* @param visitor :: Visitor to be accepted bu this actor.
-* @param rule :: A rule defining visitor acceptance by assembly actors.
-*/
-bool InstrumentActor::accept(GLActorVisitor &visitor, VisitorAcceptRule rule) {
-  bool ok = m_scene.accept(visitor, rule);
-  visitor.visit(this);
-  invalidateDisplayLists();
-  return ok;
+void InstrumentActor::setupPhysicalInstrument() {
```

Comment:
I would break this up somehow or rename. The usage implies that you are setting up a physical instrument, but actually you do nothing if `isPhysicalView()` does not return true.